### PR TITLE
process: initial documentation of verification

### PR DESCRIPTION
**Pull request by pahmann**
This is the porting of the work done in the incubator project. 

It includes the following artifacts, adopted to the latest syntax and checks for S-Core:
- Guidance
- Work Products
- Report template
- Requirement to test linking
- Test case templates
- S-Core verification plan draft

Implements: #320

Also-by: @julian711

### CODE REVIEW

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   # https://www.apache.org/licenses/LICENSE-2.0
+   #
+   # SPDX-License-Identifier: Apache-2.0
+   # *******************************************************************************
+
+Verification Guidance
+=====================
+
+This guideline outlines the responsibilities and procedures for developers performing verification activities (test
+case creation, inspection, and review) for S-Core documentation, Rust and C/C++ projects of the platform.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:19]: - should we leave out name "S-Core"?
- documentation + tooling? We also need some guidance to link our tools to process requirements for Tool Qualification

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+:ref:`sw_verification`.
+
+Test Case Development
+------------------------
+
+Following aspect should be considered when developing test cases:
+
+* **Comprehensive Coverage:** Test cases should cover all functional and non-functional requirements, including
+  positive, negative, and boundary conditions. Specific attention should be given to corner cases and error handling.
+* **Unit Testing:** Focus on isolating and testing individual units or components of the code. Strive for high code coverage.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:43]: Shall we mention something like:
- Component Test -> Requiremements
- Unit Test -> Branch + Line Coveage?
- Also specify targets for "high code coverage"?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    * **Rust:** Utilize the built-in testing framework with ``#[test]`` attributes and the ``cargo test`` command.
+    * **Python:** Use ``pytest`` frameworks.
+    * **C++:** Use Google Test frameworks.
+* **Integration Testing:** Verify the interaction between different components or modules.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:47]: Mention ITF Framework?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+* **Platform Testing:** Test the platform with configured features as a whole.
+* **Regression Testing:** Ensure that changes do not introduce new defects. Automate regression tests where possible.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:49]: Unclear, Regression testing will be done in CI?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:1]: For me this looks like more like the "concept description"
Currently I´m somehow missing the steps what I need to do when I want to start a unit test

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+       - The objective of the test.
+       - Inputs
+       - Expected outcome (e.g. "A success message is displayed.")
+       - Test environment (e.g. network configuration, clean system state)
+     -
+   * - Status
+     - valid | invalid
+     -
+     -
+   * - TestType
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:79]: Provide more examples what this means

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+     - Requirements-based test | Design based tests
+     - Assumptions of Use are treated as requirements.
+     -
+   * - DerivationTechnique
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:83]: Provide more examples + description

File: docs/process/process_areas/verification/_assets/requirements_workflow_verification.svg
Comment (hoe-jo) [inline docs/process/process_areas/verification/_assets/requirements_workflow_verification.svg:1]: Maybe take out parent req part and test suite?

File: docs/process/process_areas/verification/guidance/verification_report_template.rst
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_report_template.rst:1]: We should try to setup a more automated template where more results get generated

File: docs/platform_management_plan/software_verification.rst [hunk 7]
```diff
@@ -14,14 +14,372 @@
...
+overview of the different methods that are applicable in the project are given in this section. Usually the defined
+methods are not applied on each verification level. Due to that the following tables contain a column that defines the
+applicable level. Another column defines if a respective method is supposed to be applied if the linked references are
+QM or ASIL B relevant.
+
+Automated test cases should contain further information about which methods have been applied. The corresponding
+guidance is given here: :doc:`/process/process_areas/verification/guidance/verification_guidance`. The identifier of the respective
+method is to be used as meta data (*TestType* and *DerivationTechnique*).
+
+.. list-table:: Software verification methods (TestType)
```
Comment (aschemmel-tech) [inline docs/platform_management_plan/software_verification.rst:181]: Findings from audit still to resolve, will be checked again in second iteration

**Review by aschemmel-tech — CHANGES_REQUESTED** [review]
see inline comments, first iteration

**Comment by pahmann** [comment]
@hoe-jo this commit in #441 is for you: [workproducts: update verification and requirements](https://github.com/eclipse-score/score/pull/441/commits/9127eaec546fa7b3ca6929309657c9694c2867c7)

I removed some duplicated items and added the requirements inspection to your work products, as you already have the template for it.

File: docs/process/process_areas/verification/guidance/verification_guidance.rst (hunk 5 above)
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_guidance.rst:81]: Test type descriptions like for boundary analysis is a topic which will be discussed within testing community (and documented in process).

File: docs/process/process_areas/verification/guidance/verification_report_template.rst [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+=============================
+
+.. gd_temp:: Module Verification Report Template
+    :id: gd_temp__mod_ver_report
+    :status: valid
+    :complies: std_wp__iso26262__software_12, std_wp__iso26262__software_15, std_wp__iso26262__support_13, std_wp__iso26262__support_20, std_wp__iso26262__analysis_5, std_wp__iso26262__analysis_7, std_wp__iso26262__support_19
+
+This document implements <add "need" link>
+
+   +---------------------------+-------------------------------+
```
Comment (pahmann) [inline docs/process/process_areas/verification/guidance/verification_report_template.rst:27]: Remove table and follow new guideline on document handling as in PR.

File: docs/process/process_areas/verification/guidance/verification_report_template.rst [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+   | Document ID               | PLATFORM-VERIFICATION-REPORT  |
+   +---------------------------+-------------------------------+
+   | Project Name              | <Project>                     |
+   +---------------------------+-------------------------------+
+   | ASIL                      | B                             |
+   +---------------------------+-------------------------------+
+   | AUTHOR                    | <Committer>                   |
+   +---------------------------+-------------------------------+
+   | Reviewer                  | <Quality Manager>             |
```
[the remark this answers is not in the document]
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_report_template.rst:122]: See above. Header will be replaced to match documentation management plan requirements

File: docs/platform_management_plan/software_verification.rst [hunk 10]
```diff
@@ -14,14 +14,372 @@
...
+#. :need:`wp__verification__platform_test`
+#. :need:`wp__verification__platform_ver_report`
+#. :need:`wp__verification__feat_int_test`
+#. :need:`wp__verification__comp_int_test`
+#. :need:`wp__verification__module_ver_report`
+#. :need:`wp__verification__component_test`
+#. :need:`wp__verification__sw_unit_test`
+#. :need:`wp__sw_req_inspect`
+#. :need:`wp__sw_arch_verification`
+#. :need:`wp__sw_code_inspect`
```
[the remark this answers is not in the document]
Comment (pahmann) [reply docs/platform_management_plan/software_verification.rst:142]: It is a good point. Do we make it part of the wp__verification__sw_unit_test or should we have a dedicated wp for it?

File: docs/process/process_areas/verification/guidance/verification_process_toolreqs.rst [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+    .. code-block:: cpp
+
+      class TestSuite : public ::testing::Test{
+         public:
+            void SetUp() override
+            {
+               RecordProperty("Status", "<Status>");
+               RecordProperty("ASIL", "<ASIL>");
+               RecordProperty("Priority", "<Priority>");
```
Comment (pahmann) [inline docs/process/process_areas/verification/guidance/verification_process_toolreqs.rst:57]: @hoe-jo  Do we really want to have Priority a property? Also it is not described anywhere so far and we want to execute all tests in CI. I wonder, if we really need it.

File: docs/process/process_areas/verification/guidance/verification_report_template.rst
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_report_template.rst:1]: The template already says "Note2: All the above lists are generated automatically". Do you require more?

File: docs/process/process_areas/verification/_assets/requirements_workflow_verification.svg
Comment (pahmann) [reply docs/process/process_areas/verification/_assets/requirements_workflow_verification.svg:1]: addressed in commit [ebc7e98](https://github.com/eclipse-score/score/pull/441/commits/ebc7e98debbb9ed7fffc358f40316ca99823c8c6)

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+  Automate regression tests where possible as they will get executed as part of the CI.
+* **Performance Testing (when applicable):** Evaluate the performance characteristics of the code,
+  such as execution time, memory usage, and resource utilization.
+* **Tool Qualification Testing:** Test the platform tools based on their tool requirements to achieve tool qualification.
+
+Test specification
+------------------
+
+.. list-table:: Test specification properties
+    :header-rows: 1
```
[the remark this answers is not in the document]
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_guidance.rst]: changed to `:widths: 10 30 40 20`

File: docs/process/process_areas/verification/workflows.rst [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+   :status: valid
+   :tags: verification
+   :responsible: rl__committer
+   :approved_by: rl__technical_lead
+   :supported_by: rl__safety_manager, rl__infrastructure_tooling_community
+   :input: wp__requirements__stkh, wp__platform_mgmt
+   :output: wp__verification__plan
+   :contains: gd_guidl__verification_guide
+
+   ``[TODO: Is this the right relationship for the verification guidance?]``
```
Comment (aschemmel-tech) [inline docs/process/process_areas/verification/workflows.rst:45]: yes it is, if the gd_guidl__verification_guide explains how the Verification Plan is created.

File: docs/process/process_areas/verification/workflows.rst [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+   ``[TODO: Add additional inputs.]``
+
+   The verification plan is created and maintained by :need:`rl__committer`. It will clearly
+   outline all aspects of the verification activities, provide a roadmap for the verification
+   efforts throughout the software development lifecycle. The plan should be dynamic and updated
+   as needed throughout the project lifecycle.
+
+
+``[TODO: Additional workflows to be added]``
```
Comment (aschemmel-tech) [inline docs/process/process_areas/verification/workflows.rst:55]: resolve TODOs

File: docs/process/process_areas/verification/workproducts.rst [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+   :complies: std_wp__iso26262__support_12
+
+   Component Testing verifies Component Requirements
+
+.. workproduct:: Component Integration test
+   :id: wp__verification__comp_int_test
+   :status: valid
+   :complies: std_wp__iso26262__software_13, std_wp__iso26262__support_12
+
+   Integration Testing verifies component architecture:
```
Comment (aschemmel-tech) [inline docs/process/process_areas/verification/workproducts.rst:103]: missing testing of Detailed Design (i.e. integration od units into component)

File: docs/process/process_areas/verification/workproducts.rst [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+
+   - all interfaces from Static view and
+   - all flows from Dynamic View and performance: i.e. RAM and processor usage on reference HW
+
+.. workproduct:: Unit test
+   :id: wp__verification__sw_unit_test
+   :status: valid
+   :complies: std_wp__iso26262__software_11, std_wp__iso26262__support_12
+
+   Unit testing verifies component requirements and detailed design (traced to). Tooling defined
```
Comment (aschemmel-tech) [inline docs/process/process_areas/verification/workproducts.rst:113]: need to remove component requirements

File: docs/process/process_areas/verification/workproducts.rst [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+     * : ref : ` arch_design_process `
+       ``[make proper ref after Arch PR #500 is merged]`` implementing :need:`wp__sw_arch_verification`
+     * Detailed design implementing :need:`wp__sw_code_inspect`
+
+Tool Evaluation
+^^^^^^^^^^^^^^^
+
+As part of tool qualification as supporting function it is handled as follows
+
+     * :need:`rl__infrastructure_tooling_community` implementing :need:`wp__tool_eval`
```
Comment (aschemmel-tech) [inline docs/process/process_areas/verification/workproducts.rst:143]: recommend to put here link to "future process" or the plan document in PMP

File: docs/process/process_areas/verification/index.rst [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+-----
+
+For verification activities no additional roles need to be defined.
+
+Verification artifacts in form of test cases and specification are created by:
+
+   * :need:`rl__contributor`
+   * :need:`rl__committer`
+
+Inspection activities on requirement, architecture and detailed design are handled within these process areas.
```
Comment (aschemmel-tech) [inline docs/process/process_areas/verification/index.rst:59]: this is somehow a duplicate to what is in workproducts.rst recommend to delete one (maybe better the workproduct.rrst). And move this from "Roles" to "Workproducts" section.

Comment (pahmann) [reply docs/process/process_areas/verification/index.rst]: I put it as introductory sentence into the verification_workproducts.rst

Comment (pahmann) [reply docs/process/process_areas/verification/index.rst]: Whole roles parts is now put under verification_roles.rst and I will remove this part here.

File: docs/process/process_areas/verification/index.rst [hunk 19]
```diff
@@ -0,0 +1,48 @@
+..
```
[the remark this answers is not in the document]
Comment (pahmann) [reply docs/process/process_areas/verification/index.rst:1]: Now replicates the requirements index.rst adopted for verification.

File: docs/platform_management_plan/software_verification.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (pahmann) [reply docs/platform_management_plan/software_verification.rst:143]: I put it to sw_unit_test, but I am not really convinced. Maybe we can resolve this on Monday.

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+     - Type
+     - Description
+     - Helpful links
+   * - Verifies
+     - Sphinx-needs Ids
+     - Links to one or more requirement Ids
+     -
+   * - Description
+     - Text
+     - The description should include
```
Comment (masc2023) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:82]: Is this table correct, looks not?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst (hunk 1 above)
Comment (hoe-jo) [reply docs/process/process_areas/verification/guidance/verification_guidance.rst]: process requirements are still missing

File: docs/process/process_areas/verification/verification_concept.rst [hunk 21]
```diff
@@ -0,0 +1,201 @@
...
+
+
+General Traceability Concept
+============================
+
+To allow a traceability from any line of code to a written requirement, unit tests are linked
+to other unit tests or component tests. This linking is done using metatags. This is also true for
+component integration tests linking to the component architecture.
+
+Comoponent tests are linked to component requirements directly.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/verification_concept.rst:167]: typo: Comoponent

File: docs/process/process_areas/verification/verification_concept.rst [hunk 22]
```diff
@@ -0,0 +1,201 @@
...
+  Automate regression tests where possible as they will get executed as part of the CI.
+* **Performance Testing (when applicable):** Evaluate the performance characteristics of the code,
+  such as execution time, memory usage, and resource utilization.
+* **Tool Qualification Testing:** Test the platform tools based on their tool requirements to achieve tool qualification.
+
+
+General Traceability Concept
+============================
+
+To allow a traceability from any line of code to a written requirement, unit tests are linked
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/verification_concept.rst:163]: I would rathere state: Here only linkage to testcases and not to "any line of code"

File: docs/process/process_areas/verification/verification_concept.rst [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+requirements where requirements shall be linked to test cases on the respective level. However also a statement
+concerning the completeness of the test suite shall be generated. This means that also a linkage document shall be
+generated including:
+
+- the hash and UID of the requirement which was evaluated for test coverage
+- the UIDs of the test cases which are required to fully cover the requirement.
+
+So if the content of the requirement is altered also the hash will change making it necessary to revisit the linkage of all test cases to the requirement again.
+
+If the status of the linked test case and the linkage document is valid the attribute *testcovered* shall be set to *YES* during the Sphinx Build.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/verification_concept.rst:201]: Process Requirement?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+
+The execution of the tests is based on a full automation defined by build pipelines.
+The analysis of the test results needs to be performed by the contributor.
+
+There may be the need for limited number of manually executed test cases.
+These manually executed test cases are execution script driven, where a script guides through the
+test cases and reports the result in the same logging format as automated tests do.
+
+``[TODO: Reporting of manual execution results is pending]``
+
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:92]: Maybe add how to test locally?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+
+Reporting of failing test cases
+-------------------------------
+
+Any failing test case requires an ISSUE.
+The passing rate of safety-critical test cases need to be 100% in order to release the affected component.
+In case of a lower pass rate than 100% for QM level tests, the :need:`rl__technical_lead` and
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:99]: and?

File: docs/process/process_areas/verification/guidance/verification_guidance.rst [hunk 26]
```diff
@@ -0,0 +1,105 @@
...
+test cases and reports the result in the same logging format as automated tests do.
+
+``[TODO: Reporting of manual execution results is pending]``
+
+
+Reporting of failing test cases
+-------------------------------
+
+Any failing test case requires an ISSUE.
+The passing rate of safety-critical test cases need to be 100% in order to release the affected component.
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_guidance.rst:98]: Maybe add a defined workflow for justification?

File: docs/process/process_areas/verification/guidance/verification_specification.rst [hunk 27]
```diff
@@ -0,0 +1,137 @@
...
+
+.. list-table:: Test specification properties
+   :header-rows: 1
+   :widths: 10 30 40 20
+
+   * - Property
+     - Type / Values
+     - Description
+     - Helpful links
+   * - Verifies
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_specification.rst:38]: Shall we split verifies into two metatags: fully_verifies and partially_verifies

File: docs/process/process_areas/verification/guidance/verification_specification.rst [hunk 28]
```diff
@@ -0,0 +1,137 @@
...
+     - Sphinx-needs Ids
+     - Links to one or more requirement Ids
+     -
+   * - Description
+     - Text
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_specification.rst:43]: Somehow "Text" is missing in rendered document?
Maybe some list issue, since unordered lists are usually defined with "*" and not "-" in sphinx

File: docs/process/process_areas/verification/guidance/verification_specification.rst [hunk 29]
```diff
@@ -0,0 +1,137 @@
...
+       - Expected outcome (e.g. "A success message is displayed.")
+       - Test environment (e.g. network configuration, clean system state)
+     -
+   * - Status
+     -
+       - valid
+       - invalid
+     -
+     -
+   * - TestType
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_specification.rst:57]: Maybe align with Concept (line 95ff)

File: docs/process/process_areas/verification/guidance/verification_report_template.rst [hunk 30]
```diff
@@ -0,0 +1,124 @@
...
+      |
+      | **4.3. Manual Code Inspection**
+      | - List of components (ASIL rated) linked to inspection checklist and verdict (derived from PR export)
+      |
+      | **5. Software component qualification verification report**
+      | - Contains the needed verification results if for some components a qualification of pre-developed SW is performed.
+      |
+
+      Note1: The verification report is valid for the module version tagged together with the report
+      Note2: All the above lists are generated automatically
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_report_template.rst:83]: Note2 is in one line with Note1 in the rendered document

File: docs/process/process_areas/verification/guidance/verification_report_template.rst [hunk 31]
```diff
@@ -0,0 +1,124 @@
...
+      |
+      | **2. DFA Report**
+      | - List of the performed Feature DFA, pass/fail with open mitigations
+      |
+      | **3. Safety Analysis Report**
+      | - List of the performed Feature Safety Analysis, pass/fail with open mitigations
+      |
+
+      Note1: The verification report is valid for the platform version tagged together with the report
+      Note2: All the above lists are generated automatically
```
Comment (hoe-jo) [inline docs/process/process_areas/verification/guidance/verification_report_template.rst:124]: Note2 is in one line with Note1 in the rendered document

File: docs/process/process_areas/verification/guidance/verification_guidance.rst (hunk 25 above)
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_guidance.rst:99]: the line was lost in limbo... Now it is added.

File: docs/process/process_areas/verification/guidance/verification_specification.rst [hunk 32]
```diff
@@ -0,0 +1,137 @@
...
+  | - Username: testuser
+  | - Password: password123
+  | Expected Outcome:
+  | - User is redirected to the home page and the welcome message "Hello, testuser!" is displayed.
+
+A example for a bad test description could be:
+
+   Test user login
+
+Additional information can also be found in :doc:`./verification_guidance`.
```
Comment (pahmann) [inline docs/process/process_areas/verification/guidance/verification_specification.rst:134]: Update Link to need.

File: docs/process/process_areas/verification/guidance/verification_specification.rst [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+
+The specification is part of the test implementatin and has to comply to the requirements
```
Comment (pahmann) [inline docs/process/process_areas/verification/guidance/verification_specification.rst:136]: typo

File: docs/platform_management_plan/software_verification.rst [hunk 34]
```diff
@@ -12,16 +12,388 @@
...
+All existing test cases should be executed within continuous integration pipelines to verify initially developed
+components or software changes. A specific selection of sub sets is not planned. The fact that all existing and
+automated tests get executed continuously covers the approach to identify regressions.
+
+Work products and traceability
+------------------------------
+
+The traceability between verification relevant work products is one of the defined objectives.
+An overall overview of the different work products and their relationship is given in project
+context - see :doc:`/process/workproducts/index`.
```
Comment (pahmann) [inline docs/platform_management_plan/software_verification.rst:348]: check proper link.

File: docs/process/process_areas/verification/workproducts.rst [hunk 35]
```diff
@@ -0,0 +1,153 @@
...
+
+Inspection
+**********
+
+Inspection activities on requirement, architecture and detailed design are handled within these process areas.
+
+The work products are handled within these process areas:
+
+     * :ref:`requirements_engineering` implementing :need:`wp__requirements__inspect`
+     * : ref : ` arch_design_process `
```
Comment (pahmann) [inline docs/process/process_areas/verification/workproducts.rst:142]: resolve missing refs.

File: docs/platform_management_plan/software_verification.rst [hunk 36]
```diff
@@ -12,16 +12,423 @@
...
+- Statement/Branch/Path coverage as defined by their specific thresholds
+- Static analysis and Linting
+- :need:`wp__sw_code_inspect` for safety-critical implementation
+
+``[TODO: What can we do, what shall we do: Data Flow Coverage?]``
+
+Coverage of architectural design
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Beside :need:`wp__verification__comp_int_test` the following aspects define the coverage of the
```
Comment (pahmann) [inline docs/platform_management_plan/software_verification.rst:300]: Add feature architectural parts.

File: docs/process/process_areas/verification/guidance/verification_specification.rst (hunk 29 above)
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_specification.rst:57]: Good point. I will make a reference to the concept to check existing types.

File: docs/process/process_areas/verification/guidance/verification_specification.rst [hunk 37]
```diff
@@ -0,0 +1,137 @@
...
+       - requirements-based
+       - interface-test
+       - boundary
+       - coverage (various types apply, shall be tool supported)
+       - for :need:`wp__verification__sw_unit_test` also fault-injection
+     -
+       - Requirements-based test (Assumptions of Use are treated as requirements.)
+       - Design based tests
+
+       Consult the implementation of :need:`wp__verification__plan` for the full list of allowed types.
```
Comment (pahmann) [inline docs/process/process_areas/verification/guidance/verification_specification.rst:68]: Put this in front of table as explanation. Or below, but not part of table.

File: docs/process/process_areas/verification/guidance/verification_specification.rst (hunk 28 above)
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_specification.rst:43]: For me it renders:
image

**Comment by pahmann** [comment]
@aschemmel-tech kindly do a review on the current state. I should have addressed all your pending concerns.

Missing part for final commit on this PR will be the transfer of TODOs into improvement issue(s).

File: docs/process/process_areas/verification/verification_concept.rst [hunk 38]
```diff
@@ -0,0 +1,205 @@
...
+
+   * Set the scope of platform and testing
+   * Status reporting of verification activities
+   * Defining releae content
+   * Judge the releasable software state in case of failing test cases.
+
+     This may need involvment of :need:`rl__project_lead`,
+     :need:`rl__safety_manager`, and :need:`rl__security_manager`
+
+#. :need:`SW developer/Archticts <rl__contributor>`
```
Comment (masc2023) [inline docs/process/process_areas/verification/verification_concept.rst:52]: typo Archticts -> Architects

File: docs/process/process_areas/verification/verification_concept.rst [hunk 39]
```diff
@@ -0,0 +1,205 @@
...
+===========
+
+#. :need:`Technical Lead <rl__technical_lead>`
+
+   * Set the scope of platform and testing
+   * Status reporting of verification activities
+   * Defining releae content
+   * Judge the releasable software state in case of failing test cases.
+
+     This may need involvment of :need:`rl__project_lead`,
```
Comment (masc2023) [inline docs/process/process_areas/verification/verification_concept.rst:49]: typo involment -> involvement

File: docs/process/process_areas/verification/verification_concept.rst [hunk 40]
```diff
@@ -0,0 +1,205 @@
...
+
+#. :need:`Technical Lead <rl__committer>`
+
+   * Set the scope of platform and testing
+   * Status reporting of verification activities
+   * Defining releae content
+
+#. :need:`Infrastructure/Tooling Developer <rl__infrastructure_tooling_community>`
+
+   * Enalbes execution of test cases in CI
```
Comment (masc2023) [inline docs/process/process_areas/verification/verification_concept.rst:75]: typo Enalbes -> Enables

File: docs/process/process_areas/verification/verification_concept.rst [hunk 41]
```diff
@@ -0,0 +1,205 @@
...
+   * Provides tools for test generation
+   * Integrates static analysis, linting, test frameworks into CI
+
+#. :need:`Safety Manager <rl__safety_manager>`
+
+   * Supports the verification activities for safety-critical work products.
+
+#. :need:`External Auditor <rl__external_auditor>`
+
+   * understand activities, planning, processes definition, and execution reports for verifcation activities
```
Comment (masc2023) [inline docs/process/process_areas/verification/verification_concept.rst:86]: Start with Understand, as above

File: docs/process/process_areas/verification/verification_concept.rst [hunk 42]
```diff
@@ -0,0 +1,205 @@
...
+
+#. "Distributor" (external role)
+
+   * reexecution of test cases
```
Comment (masc2023) [inline docs/process/process_areas/verification/verification_concept.rst:90]: Start with Reexecution as above

File: docs/process/process_areas/verification/verification_concept.rst [hunk 43]
```diff
@@ -0,0 +1,205 @@
...
+Automated test cases should contain further information about which methods have been applied.
+The corresponding guidance is given here: :need:`gd_guidl__verification_guide`.
+The identifier of the respective method is to be used as meta data (*TestType* and *DerivationTechnique*).
+
+Test Case Development
+=====================
+
+Following aspect should be considered when developing test cases:
+
+* **Comprehensive Coverage:** Test cases should cover all functional and non-functional requirements, including
```
Comment (masc2023) [inline docs/process/process_areas/verification/verification_concept.rst:138]: do we really cover non-functional requirements, like process relevant topics?

File: docs/process/process_areas/verification/workflows.rst (hunk 14 above)
Comment (pahmann) [reply docs/process/process_areas/verification/workflows.rst]: Remaining ToDos which where spread around in the whole verification process are now all collected in issue #623

File: docs/process/process_areas/verification/guidance/verification_guidance.rst (hunk 24 above)
Comment (pahmann) [reply docs/process/process_areas/verification/guidance/verification_guidance.rst]: Added this finding to issue #623

**Comment by pahmann** [comment]
@aschemmel-tech, from my perspective we are ready to merge. 🎉

File: docs/platform_management_plan/software_verification.rst [hunk 44]
```diff
@@ -14,14 +14,372 @@
...
+#. :need:`wp__verification__sw_unit_test`
+#. :need:`wp__sw_req_inspect`
+#. :need:`wp__sw_arch_verification`
+#. :need:`wp__sw_code_inspect`
+
+
+As part of supporting activities:
+
+#. :need:`wp__verification__plan`
+#. :need:`wp__verification__specification`
```
[the remark this answers is not in the document]
Comment (aschemmel-tech) [reply docs/platform_management_plan/software_verification.rst]: agreed in working meeting Mar-10 to remove

File: docs/process/process_areas/verification/verification_concept.rst (hunk 43 above)
Comment (aschemmel-tech) [reply docs/process/process_areas/verification/verification_concept.rst]: yes, these are also ones from external compoents AoUs

**Review by aschemmel-tech — APPROVED** [review]
Agree, let's merge
